src/coll/mod.rs

```diff
@@ -1501,3 +1501,23 @@ impl Serialize for Namespace {
         serializer.serialize_str(&(self.db.clone() + "." + &self.coll))
     }
 }
+
+impl FromStr for Namespace {
+    type Err = Error;
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let mut parts = s.split('.');
+
+        let db = parts.next();
+        let coll = parts.collect::<Vec<_>>().join(".");
+
+        match (db, coll) {
+            (Some(db), coll) if !coll.is_empty() => std::result::Result::Ok(Self {
```

Comment:
You can just use `Ok` here instead of `std::result::Result::Ok` (and likewise for `std::result::Result::Err` below) - `Ok` and `Err` are common enough that they're part of the [Rust prelude](https://doc.rust-lang.org/std/prelude/index.html).